### DISCUSSION TITLE
Dropdown Menu

### DISCUSSION THREAD

**Discussion by PotentialAviator15** [Pycord-Development/pycord#406] [Q&A] [answered]
I've tried the dropdown menu, and I just started coding python 4 months ago 😅 how do I use the dropdown menu for adding roles?

I got the dropdown menu to work when I copied from the example file, but I don't know how to utilize it to add roles. I know what section I need to change, but I'm not entirely sure what I will be doing with the new code that I have to put in for the roles to work.

**Comment by esmaycat** [comment, marked as answer]
If you want to add roles when someone selects an item in your select menu, in your callback check which item(s) they have selected with `select.values` or `self.values` if you are subclassing `discord.ui.Select`. Then you can add a role to `interaction.user` (the person who selected the item) by checking what they selected.

**Reply by PotentialAviator15** [reply]
I will need to play around a little bit because I think I can understand what you are saying, but I need to be sure. Thanks!

**Reply by pratheek78** [reply]
You could also assign each option a value and then make if statements for when the user reacts with the options.

**Reply by PotentialAviator15** [reply]
I find that the easier option @pratheek78

**Reply by pratheek78** [reply]
Also, you could make another check to make sure only the author can interact with the dropdown
